frontend/controller/service-worker.js

```diff
@@ -199,61 +198,51 @@ sbp('sbp/selectors/register', {
       }
 
       const permissionState = await registration.pushManager.permissionState({ userVisibleOnly: true })
+      const granted = permissionState === 'granted' || Notification.permission === 'granted'
+      console.info(`[service-worker/setup-push-subscription] setup: notifications (${notificationEnabled}) perms (${granted})`)
 
-      // Safari sometimes incorrectly reports 'prompt' when using
-      // `registration.pushManager.permissionState`.
-      const existingSubscription = permissionState === 'granted' || Notification.permission === 'granted'
-        ? await registration.pushManager.getSubscription().then((subscription) => {
-          if (
-            !subscription ||
-            (subscription.expirationTime != null && subscription.expirationTime <= Date.now())
-          ) {
-            console.info('[service-worker/setup-push-subscription] attempting to create a new subscription', subscription)
-            return sbp('push/getSubscriptionOptions').then(function (options) {
-              return registration.pushManager.subscribe(options)
-            })
-          } else {
-            const subClone = cloneDeep(subscription)
-            if (subscription?.endpoint) {
-              // hide the full endpoint from the logs for privacy
-              subClone.endpoint = new URL(subClone.endpoint).host
-            }
-            console.info('[service-worker/setup-push-subscription] got existing subscription:', subClone)
+      // Safari sometimes incorrectly reports 'prompt' when using `registration.pushManager.permissionState`
+      let subscription = null
+      let subPrivVer = subscription
+      if (notificationEnabled && granted) {
+        try {
+          subscription = await registration.pushManager.getSubscription()
+          let newSub = false
+          if (!subscription || (subscription.expirationTime != null && subscription.expirationTime <= Date.now())) {
+            subscription = await registration.pushManager.subscribe(await sbp('push/getSubscriptionOptions'))
+            newSub = true
+          }
+          subPrivVer = cloneDeep(subscription)
+          if (subscription?.endpoint) {
+            // hide the full endpoint from the logs for privacy
+            subPrivVer.endpoint = new URL(subPrivVer.endpoint).host
```

Comment:
Not exactly sure what these changes here are doing, other than some refactoring (I prefer things mostly like they were, because it was written to use immutable (`const`) variables, which is IMO easier to reason about) and removing the retry logic.

Reply:
> In the case of removing the retry logic, I'm not entirely sure why you want to do that

Because at this point in time there shouldn't be a failure (because of the other refactoring changes make sure that `'service-worker/setup-push-subscription'` gets called after `LOGIN_COMPLETE`), and if there is now the `gi.ui/prompt` will be shown in more situations.

Reply:
>  there shouldn't be a failure

But there can be, and retrying seems like the reasonable thing to do here.

Reply:
Retrying will automatically happen any time the user refreshes the page or goes offline and back online, or toggles the notifications switch.